compiler-core/templates/prelude.mjs

```diff
@@ -201,8 +239,23 @@ export function codepointBits(codepoint) {
 }
 
 // @internal
-export function float64Bits(float) {
-  return new Uint8Array(Float64Array.from([float]).buffer).reverse();
+export function sizedFloat(float, size, isBigEndian) {
+  if (size !== 32 && size !== 64) {
```

Comment:
Let's use a bool as that forces it to be one or the other.

Reply:
I don't think that's an option because the size can be specified dynamically:

```gleam
  let a = int.random(1000)
  let b = <<1.0:float-size(a)>>
```

Both Erlang and JS throw an error if the size at runtime isn't supported for floats.